misc/docker-integration-test.py

```diff
@@ -0,0 +1,114 @@
+
+
+def fake_vendor():
+    """
+    "vendors" notary into docker by copying all of notary into the docker
+    vendor directory - also appending several lines into the Dockerfile because
+    it pulls down notary from github and builds the binaries
+    """
+    docker_notary_relpath = "vendor/src/github.com/docker/notary"
+    docker_notary_abspath = os.path.join(DOCKER_DIR, docker_notary_relpath)
+
+    print("copying notary ({0}) into {1}".format(NOTARY_DIR, docker_notary_abspath))
+
+    def ignore_dirs(walked_dir, _):
+        """
+        Don't vendor everything, particularly not the docker directory
+        recursively, if it happened to be in the notary directory
+        """
+        if walked_dir == NOTARY_DIR:
+            return [".git", ".cover", "docs", "bin"]
+        elif walked_dir == os.path.join(NOTARY_DIR, "fixtures"):
+            return ["compatibility"]
+        return []
+
+    if os.path.exists(docker_notary_abspath):
+        shutil.rmtree(docker_notary_abspath)
+    shutil.copytree(
+        NOTARY_DIR, docker_notary_abspath, symlinks=True, ignore=ignore_dirs)
+
+    # hack this because docker/docker's Dockerfile checks out a particular version of notary
+    # based on a tag or SHA, and we want to build based on what was vendored in
+    dockerfile_addition = ("\n"
+        "RUN set -x && "
+        "GOPATH=$(pwd)/vendor/src/github.com/docker/notary/Godeps/_workspace:$GOPATH "
+        "go build -o /usr/local/bin/notary-server github.com/docker/notary/cmd/notary-server &&"
+        "GOPATH=$(pwd)/vendor/src/github.com/docker/notary/Godeps/_workspace:$GOPATH "
+        "go build -o /usr/local/bin/notary github.com/docker/notary/cmd/notary")
+
+    with open(os.path.join(DOCKER_DIR, "Dockerfile")) as dockerfile:
+        text = dockerfile.read()
+
+    if not text.endswith(dockerfile_addition):
+        with open(os.path.join(DOCKER_DIR, "Dockerfile"), 'a+') as dockerfile:
+            dockerfile.write(dockerfile_addition)
+
+    # hack the makefile so that we tag the built image as something else so we
+    # don't interfere with any other docker test builds
+    with open(os.path.join(DOCKER_DIR, "Makefile"), 'r') as makefile:
+        makefiletext = makefile.read()
+
+    with open(os.path.join(DOCKER_DIR, "Makefile"), 'wb') as makefile:
+        image_name = os.getenv("DOCKER_TEST_IMAGE_NAME", "notary-docker-vendor-test")
+        text = re.sub("^DOCKER_IMAGE := .+$", "DOCKER_IMAGE := {0}".format(image_name),
+                      makefiletext, 1, flags=re.M)
+        makefile.write(text)
+
+def run_integration_test():
+    """
+    Presumes that the fake vendoring has already happened - this runs the
+    integration tests.
+    """
+    env = os.environ.copy()
+    env["TESTFLAGS"] = '-check.f DockerTrustSuite*'
+    subprocess.check_call(
+        "make test-integration-cli".split(), cwd=DOCKER_DIR, env=env)
+
+if __name__ == "__main__":
+    if len(sys.argv) > 1:
```

Comment:
If we don't accept any arguments, maybe just print a warning but continue? Don't feel strongly, just my preferred behaviour in these kinds of cases. Plenty of cases doing it both ways (halting vs ignoring unused args).